src/modules/75-cd/components/EnvironmentGroups/CreateEnvironmentGroupModal.tsx

```diff
@@ -0,0 +1,417 @@
+  }, [])
+
+  const onSubmit = async (value: EnvironmentGroupResponseDTO) => {
+    setCreateLoading(true)
+    try {
+      const values = cleanData(value)
+      const body = yamlStringify(
+        JSON.parse(
+          JSON.stringify({
+            environmentGroup: { ...values }
+          })
+        )
+      )
+      const response = await createEnvironmentGroupPromise({
+        body,
+        queryParams: {
+          accountIdentifier: accountId,
+          orgIdentifier,
+          projectIdentifier
+        },
+        requestOptions: { headers: { 'Content-Type': 'application/yaml' } }
+      })
+
+      if (response.status === 'SUCCESS') {
+        clear()
+        showSuccess(getString('common.environmentGroup.created'))
+        history.push(
+          routes.toEnvironmentGroupDetails({
+            orgIdentifier,
+            projectIdentifier,
+            accountId,
+            module,
+            environmentGroupIdentifier: defaultTo(response.data?.envGroup?.identifier, ''),
+            sectionId: EnvironmentGroupDetailsTab.ENVIRONMENTS
+          })
+        )
+      } else {
+        throw response
+      }
+    } catch (e: any) {
+      showError(getErrorInfoFromErrorObject(e, true))
+    }
+    setCreateLoading(false)
+  }
+
+  const handleModeSwitch = useCallback(
+    (view: SelectedView) => {
+      if (view === SelectedView.VISUAL) {
+        const yaml = defaultTo(yamlHandler?.getLatestYaml(), '')
+        const yamlVisual = parse(yaml).environmentGroup as EnvironmentGroupResponseDTO
+
+        if (yamlVisual) {
+          formikRef.current?.setValues({
+            ...(cleanData(yamlVisual) as EnvironmentGroupResponseDTO)
+          })
+        }
+      }
+
+      setSelectedView(view)
+    },
+    [yamlHandler?.getLatestYaml]
+  )
+
+  const { data: environmentGroupSchema } = useGetYamlSchema({
+    queryParams: {
+      entityType: 'EnvironmentGroup',
+      projectIdentifier,
+      orgIdentifier,
+      accountIdentifier: accountId,
+      scope: getScopeFromDTO({ accountIdentifier: accountId, orgIdentifier, projectIdentifier })
+    }
+  })
+
+  const queryParams = useMemo(
+    () => ({
+      accountIdentifier: accountId,
+      orgIdentifier,
+      projectIdentifier,
+      size: pageSize.current,
+      searchTerm
+    }),
+    [accountId, orgIdentifier, projectIdentifier, searchTerm]
+  )
+
+  const {
+    items: environments,
+    error: fetchEnvironmentsError,
+    fetching: fetchingEnvironments,
+    attachRefToLastElement,
+    offsetToFetch
+  } = useInfiniteScroll({
+    getItems: options => {
+      return getEnvironmentListPromise({
+        queryParams: { ...queryParams, size: options.limit, page: options.offset }
+      })
+    },
+    limit: pageSize.current,
+    loadMoreRef,
+    searchTerm
+  })
+
+  const isFetchingEnvironmentsFirstTime = useMemo(() => {
+    return fetchingEnvironments && offsetToFetch.current === 0
+  }, [fetchingEnvironments, offsetToFetch.current])
```

Comment:
Codacy has a fix for the issue: [React Hook useMemo has a missing dependency: 'offsetToFetch'. Either include it or remove the dependency array. Mutable values like 'offsetToFetch.current' aren't valid dependencies because mutating them doesn't re-render the component.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232281)
```suggestion
  }, [fetchingEnvironments, offsetToFetch])
```